pyretailscience/plots/line.py

```diff
@@ -0,0 +1,114 @@
+If a datetime column is passed or more complex temporal plotting is needed, a warning will suggest using the **`timeline`** module, which is specifically designed for working with temporal data and performing time-based manipulation.
+- **Pre-Aggregated Data Required**: The module does not perform any data aggregation, so all data must be pre-aggregated before being passed in for plotting.
+
+"""
+
+import warnings
+
+import pandas as pd
+from matplotlib.axes import Axes, SubplotBase
+
+import pyretailscience.style.graph_utils as gu
+from pyretailscience.style.tailwind import get_base_cmap
+
+
+def plot(
+    df: pd.DataFrame,
+    value_col: str | list[str],
+    x_label: str,
+    y_label: str,
+    title: str,
+    x_col: str | None = None,
+    group_col: str | None = None,
+    legend_title: str | None = None,
+    ax: Axes | None = None,
+    source_text: str | None = None,
+    move_legend_outside: bool = False,
+    **kwargs: dict[str, any],
+) -> SubplotBase:
+    """Plots the `value_col` over the specified `x_col` or index, creating a separate line for each unique value in `group_col`.
+
+    Args:
+        df (pd.DataFrame): The dataframe to plot.
+        value_col (str or list of str): The column(s) to plot.
+        x_label (str): The x-axis label.
+        y_label (str): The y-axis label.
+        title (str): The title of the plot.
+        x_col (str, optional): The column to be used as the x-axis. If None, the index is used.
+        group_col (str, optional): The column used to define different lines.
+        legend_title (str, optional): The title of the legend.
+        ax (Axes, optional): Matplotlib axes object to plot on.
+        source_text (str, optional): The source text to add to the plot.
+        move_legend_outside (bool, optional): Move the legend outside the plot.
+        **kwargs: Additional keyword arguments for Pandas' `plot` function.
+
+    Returns:
+        SubplotBase: The matplotlib axes object.
+    """
+    if x_col is not None and pd.api.types.is_datetime64_any_dtype(df[x_col]):
+        warnings.warn(
+            f"The column '{x_col}' is datetime-like. Consider using the 'timeline' module for time-based plots.",
+            UserWarning,
+            stacklevel=2,
+        )
+    colors = get_base_cmap()
+
+    if group_col is not None:
+        pivot_df = df.pivot(index=x_col if x_col is not None else None, columns=group_col, values=value_col)
+    else:
+        pivot_df = df.set_index(x_col if x_col is not None else df.index)[value_col]
+
+    ax = pivot_df.plot(
+        ax=ax,
+        linewidth=3,
+        color=colors.colors[: len(pivot_df.columns) if group_col is not None else 1],
+        legend=(group_col is not None),
+        **kwargs,
+    )
+
+    ax = gu.standard_graph_styles(
+        ax,
+        title=title,
+        x_label=x_label,
+        y_label=y_label,
+    )
```

Comment:
Since this is now just passing parameters
```suggestion
    ax = gu.standard_graph_styles(ax, title=title, x_label=x_label,  y_label=y_label)
```